app/models/question.rb

```diff
@@ -34,6 +34,10 @@ def self.with_basic_association
     by_date.includes(user: [:social_providers])
   end
 
+  def self.from_user_subscription(user_tags)
+    joins(:tags).where(tags: { name: user_tags })
```

Comment:
My thought :thought_balloon:  on empty results would be to redirect the user to the _all questions route_. Arguable, if there shouldn't be a tag with at least one question (unless it's deleted), then, any user with empty result, in all likelyhood, have refused to subscribe to any tag.

Could you clarify on the 3 db calls you mentioned. Here's a sample query print out of the method:
```sql
SELECT "questions".* FROM "questions" INNER JOIN "resource_tags" ON "resource_tags"."taggable_id" = "questions"."id" AND "resource_tags"."taggable_type" = ? INNER JOIN "tags" ON "tags"."id" = "resource_tags"."tag_id" WHERE 1=0  [["taggable_type", "Question"]]
```
The only extraneous db query I seem to notice is that to retrieve the **user tag**. We could eliminate this by _side loading_ user tags when setting up **current_user**